jcli/build.rs

```diff
@@ -0,0 +1,82 @@
+use std::env::consts::{ARCH, OS};
+use std::process::Command;
+
+#[cfg(debug_assertions)]
+const BUILD_TYPE: &'static str = "debug";
+#[cfg(not(debug_assertions))]
+const BUILD_TYPE: &'static str = "release";
+
+fn main() {
+    let version_string = format!(
+        "{} ({}-{}{}, {}, {} [{}]) - [{}]",
+        env!("CARGO_PKG_VERSION"),
+        git_branch_name(),
+        git_commit_hash(),
+        if git_working_tree_isclean() { "" } else { "+" },
+        BUILD_TYPE,
+        OS,
+        ARCH,
+        compiler_version()
+    );
+
+    println!("cargo:rustc-env=CARGO_PKG_VERSION={}", version_string);
+}
+
+fn compiler_version() -> String {
```

Comment:
Yes, `rustc_version` crate is much better. compiler version was put as an example and probably is not needed and can be removed since the documentation already states that:

> Currently the minimum supported version of the rust compiler is 1.35, however we recommend to use the most recent stable version of the rust compiler.

Reply:
If we assume that compiler is never an issue then we can remove it, but I think that it sometimes might be useful, especially if somebody uses nightly compiler from specific date. If it's not a burden, I'd keep it.